### ISSUE TITLE
[`Pylint`] [preview] `PLE4703` false negative on any reassignment

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#18818]
### Summary

If there is any reassignment of the target variable, even when both are as a set, [modified-iterating-set (PLE4703)](https://docs.astral.sh/ruff/rules/modified-iterating-set/#modified-iterating-set-ple4703) will have a false negative. [playground](https://play.ruff.rs/9d10264a-9541-4f70-bc20-3483c22ce9b4)
```py
works = {1, 2, 3}
for num in works:  # PLE4703
    works.add(num + 5)

nums = {1, 2, 3}
nums = {1, 2, 3}
for num in nums:   # No error
    nums.add(num + 5)
```

### Version

playground

**Comment by MichaReiser** [comment]
Same as for https://github.com/astral-sh/ruff/issues/18855

Simply using `resolve_name` over `only_binding` trades false negatives with false positives. 

I think the right approach here is to modify the implementation to resolve all visible bindings and verify that all of them are dictionaries. I can't say how hard that is because I'm not very familiar with our semantic model. 

An alternative is to change `only_binding` to return `Some` even if there are multiple bindings but if only one binding is visible unconditionally (not dependent on control flow paths)
